src/components/datepicker/js/datepickerDirective.js

```diff
@@ -291,12 +278,23 @@
     this.isFocused = false;
 
     /** @type {boolean} */
-    this.isDisabled;
-    this.setDisabled($element[0].disabled || angular.isString($attrs.disabled));
+    this.isDisabled = false;
+    this.setDisabled($element[0].disabled || $attrs.hasOwnProperty('disabled'));
 
-    /** @type {boolean} Whether the date-picker's calendar pane is open. */
+    /** @type {boolean} Whether the date-picker's calendar pane is open. Used internally. */
     this.isCalendarOpen = false;
 
+    /**
+     * Whether calendar should be open. Used when triggering the calendar externally.
+     * It needs to be separate from `isCalendarOpen` in order to avoid some mixups
+     * where it might trigger the calendar a couple of times in a row.
+     * @type {boolean}
+     */
+    this.isOpen = false;
+
+    /** @type {boolean} Used to prevent infinite loops when using mdOpenOnFocus. */
+    this.preventInputFocus = false;
```

Comment:
Yes, this is also a private property. When the calendar closes, it restores focus to the element that triggered it to open. With `mdOpenOnFocus` that would cause the calendar to open up again since it refocuses the input. We use this flag to prevent such cases. It used to be possible to do this by delaying the close handler, but it got a little trickier with `mdPanel` since it's also async.